templates/etc/netplan/config.yaml.j2

```diff
@@ -0,0 +1,27 @@
+network:
+  ethernets:
+{% for interface in vm_unique_net_interfaces %}
+    {{ interface.name }}:
+      addresses:
+{% if (interface.ip4 is defined) %}
+        - {{ interface.ip4 }}/{{ interface.netmask4 }}
+{% endif %}
+{% if (interface.ip6 is defined) %}
+        - {{ interface.ip6 }}/{{ interface.netmask6 }}
+{% endif %}
+{% if (interface.ip4 is defined) %}
+      gateway4: {{ interface.gw4 }}
+{% endif %}
+{% if (interface.ip6 is defined) %}
+      gateway6: {{ interface.gw6 }}
+{% endif %}
+{% if (vm_unique_net_nameserver is defined) %}
```

Comment:
Here, don't you think it would be better to use the structure that already exists (`vm_unique_net_interfaces`) instead of the static variable `vm_unique_net_nameserver`?

for example something like that:
```yaml
vm_unique_net_interfaces:
   - name: eth0
     ip4: 192.168.33.10
     gw4: 192.168.33.1
     netmask4: 24
     ip6: 200:db8::10
     gw6: 200:db8::1
     netmask6: 64
     nameservers:
       - '1.1.1.1'
       - '8.8.8.8'
```